pywbem/cim_operations.py

```diff
@@ -1941,11 +1944,16 @@ def infer_type(obj):
             elif isinstance(obj, (CIMClass, CIMInstance)):
                 return 'string'
             elif isinstance(obj, list):
-                return infer_type(obj[0]) if obj else None
+                return infer_type(obj[0], param_name) if obj else None
             elif obj is None:
                 return None
+            if isinstance(obj, int):
+                hint = " (use a CIM integer type such as pywbem.Uint32)"
+            else:
+                hint = ""
             raise TypeError(
-                _format("Unsupported parameter type {0}", type(obj)))
+                _format("Method parameter {0!A} has unsupported type {1}{2}",
```

Comment:
Suggest rewording this to something like "Method Parameter {0!A} type {1} cannot be used to infer a valid CIM data type{2}.

Reply:
DONE - changed text to:
"Method parameter {0!A} has type {1} which cannot be used to infer a valid CIM data type{2}"